packages/tooling/fast-tooling/src/data-utilities/registered-action.ts

```diff
@@ -0,0 +1,44 @@
+export interface RegisteredActionCallbackConfig {
+    /**
+     * The unique identifier
+     */
+    id: string;
+}
+
+export interface RegisteredActionConfig extends RegisteredActionCallbackConfig {
+    /**
+     * The action to take when the keycodes have been pressed
+     */
+    action: () => void;
+}
+
+/**
+ * This abstract class is for actions intended to be part
+ * of a registered class
+ */
+export abstract class RegisteredAction<C, M> {
+    private action: (config: C | RegisteredActionCallbackConfig) => void;
```

Comment:
Is `C` truly generic (could be any type, `null` for instance) or should it always be of at *least* type `RegisteredActionCallbackConfig`?

Reply:
`C` is not truly generic, it really should be any additional interface that can be combined with `RegisteredActionCalbackConfig`, so `null` would be erroneous. What solution would be best?

Reply:
I think you want somthing like the following then:

```
export abstract class RegisteredAction<C extends RegisteredActionCallbackConfig, M> {
  private action: (config: C) => void;
}
```
And maybe something similar for `M`? I'm not sure...

Reply:
Hmm, going to have to take a further look at this... That doesn't quite work because I'm using an exclusionary version to execute on the child class.

Reply:
I'm using the same method react uses and indicating that these generics should be objects with <C = {}, M = {}>, there wasn't a good way to extrapolate this out since it's not used as an extension of another interface.